dashboard/app/models/user.rb

```diff
@@ -467,6 +467,14 @@ def self.from_omniauth(auth, params)
       user.email = auth.info.email
       user.user_type = params['user_type'] || auth.info.user_type || User::TYPE_STUDENT
 
+      if auth.provider == :the_school_project
```

Comment:
If we get a lot more of these (getting user metadata in product-specific ways from Clever, The School Project, etc) it might be worth refactoring the provider/product-specific code out somehow. I don't think the camel's back broke this time, but it probably does on the next auth provider like this.